analysis_driver/driver.py

```diff
@@ -61,18 +61,22 @@ def demultiplexing_pipeline(dataset):
     app_logger.info('Job dir: ' + job_dir)
 
     run_info = reader.RunInfo(input_run_folder)
+    has_barcode = True
 
     ntf.start_stage('setup')
     reader.transform_sample_sheet(input_run_folder)
     sample_sheet = reader.SampleSheet(os.path.join(input_run_folder, 'SampleSheet_analysis_driver.csv'))
-    if not sample_sheet.validate(run_info.mask):
+    validation_results = sample_sheet.validate(run_info.mask)
+    if not validation_results:
         raise AnalysisDriverError('Validation failed. Check SampleSheet.csv and RunInfo.xml.')
-
-    mask = sample_sheet.generate_mask(run_info.mask)
+    if validation_results == 'barcodeless':
+        has_barcode = False
+        reader._remove_samplesheet_barcodes(input_run_folder)
+    mask = sample_sheet.generate_mask(run_info.mask, validation_results)
     app_logger.info('bcl2fastq mask: ' + mask)  # e.g: mask = 'y150n,i6,y150n'
```

Comment:
barcodeless value could be hidden in the SampleSheet class and accessed in the Crawlers